### DISCUSSION TITLE
Why is the server cache 2 times the browser cache

### DISCUSSION THREAD

**Discussion by rickstaa** [anuraghazra/github-readme-stats#3258] [Q&A] [answered]
I was asked why I set the server cache twice the browser cache:

https://github.com/anuraghazra/github-readme-stats/blob/2ee803fc16af70843c4a4ce37135d1ea699109b7/api/index.js#L68-L73

I will use this discussion post to explain my reasoning.

**Comment by rickstaa** [comment, marked as answer]
I use CACHE/2 in the browser for the following specific scenario:

1. Imagine one user requests a card at 12:00. In this scenario, the user's browser cache is configured to retain content for 2 hours, while the CDN cache is set to 4 hours.

2. Now, consider another user who visits the same card 2 hours before the server cache (CDN cache) expires. This action extends their browser cache period to 4 hours.

3. If this second user returns after 2 hours when the server cache has already expired, they will receive a fresh card from the server.

However, if both the browser cache and server cache are set to 4 hours, this second user won't receive a new card upon returning after the server cache expiration. This is because their browser cache is still active for another 2 hours.

This caching strategy ensures that users receive up-to-date content when their browser cache expires while leveraging longer CDN cache periods for better performance and reduced server load. @anuraghazra, @qwerty541 since this is quite an edge case, we can also simplify the code and set both the server and browser cache to the same period 👍🏻.

**Reply by qwerty541** [reply]
@rickstaa Thanks for the explanation. I'm okay with current caching strategy, i just want it to be documented somewhere.
